File: my-folder/0242-valid-anagram/solution.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        hashmap = {}
        if len(s) != len(t):
            return False
        for char in s:
            if char in hashmap.keys():
                hashmap[char] = hashmap[char] + 1
            else:
                hashmap[char] = 1
        for char in t:
            if char in hashmap.keys():
                hashmap[char] = hashmap[char] - 1
                if hashmap[char] < 0:
                    return False
            else:
                return False
        return True

File: my-folder/0242-valid-anagram/solution.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        char[] charArr1 = s.toCharArray();
        char[] charArr2 = t.toCharArray();
        HashMap<Character, Integer> map = new HashMap<>();
        for (char c : charArr1) {
            if (map.containsKey(c)) {
                map.replace(c, map.get(c) + 1);
            } else {
                map.put(c, 1);
            }
        }
        for (char c : charArr2) {
            if (map.containsKey(c)) {
                map.replace(c, map.get(c) - 1);
                if (map.get(c) == 0) {
                    map.remove(c);
                }
            } else {
                return false;
            }
        }
        return map.isEmpty();
    }
}
